monai/transforms/spatial/functional.py

```diff
@@ -160,3 +167,270 @@ def spatial_resample(
         img = img.reshape(full_shape)
     out = convert_to_tensor(img, track_meta=get_track_meta(), dtype=torch.float32)
     return out.copy_meta_from(meta_info) if isinstance(out, MetaTensor) else out  # type: ignore
+
+
+def orientation(img, original_affine, spatial_ornt, transform_info):
+    spatial_shape = img.peek_pending_shape() if isinstance(img, MetaTensor) else img.shape[1:]
+    xform = nib.orientations.inv_ornt_aff(spatial_ornt, spatial_shape)
+    img = convert_to_tensor(img, track_meta=get_track_meta())
+
+    spatial_ornt[:, 0] += 1  # skip channel dim
+    spatial_ornt = np.concatenate([np.array([[0, 1]]), spatial_ornt])
+    axes = [ax for ax, flip in enumerate(spatial_ornt[:, 1]) if flip == -1]
+    full_transpose = np.arange(len(spatial_shape) + 1)  # channel-first array
+    full_transpose[: len(spatial_ornt)] = np.argsort(spatial_ornt[:, 0])
+    extra_info = {"original_affine": original_affine}
+
+    shape_np = convert_to_numpy(spatial_shape, wrap_sequence=True)
+    shape_np = shape_np[[i - 1 for i in full_transpose if i > 0]]
+    meta_info = TraceableTransform.track_transform_meta(
+        img,
+        sp_size=shape_np,
+        affine=xform,
+        extra_info=extra_info,
+        orig_size=spatial_shape,
```

Comment:
Hi @wyli , here `spatial_shape` does not use the same rule as `spatial_resample` does to ensure that the shape is <= 3 dims. Is the rule missing or there are some other considerations? It may impact the `peek_pending_shape`

Reply:
we can remove the test case of more than 3d here, I don't think we have a use case in general for 5d affine matrix

Reply:
I see. How about 1d cases?
It seems lazy resampling only handle 2d and 3d cases? It seems `apply_transforms` has the limitation. (see: [here](https://github.com/Project-MONAI/MONAI/blob/ab800d8413df5680161ea00fb3d6c1a7aa8dd04b/monai/transforms/lazy/utils.py#L76) and [here](https://github.com/Project-MONAI/MONAI/blob/ab800d8413df5680161ea00fb3d6c1a7aa8dd04b/monai/transforms/lazy/utils.py#L41))

Reply:
Update: we've discussed and in `peek_pending_affine`, a shape check should be added and give informative warnings if the shape is not supported. I'll address it within the PR: https://github.com/Project-MONAI/MONAI/pull/6080/files